src/app/services/storage.service.ts

```diff
@@ -2,25 +2,35 @@ import { Injectable } from '@angular/core';
 import { AngularFireAuth } from '@angular/fire/auth';
 import { AngularFireStorage } from '@angular/fire/storage';
 import { User } from '../models/user';
+import { BehaviorSubject } from 'rxjs';
+import { ToastrService } from 'ngx-toastr';
 
 @Injectable({
   providedIn: 'root'
 })
 export class StorageService {
   user: User;
+  isUploadedSource: BehaviorSubject<string> = new BehaviorSubject('');
 
-  constructor(private firebaseAuth: AngularFireAuth, private firebaseStorage: AngularFireStorage) {
+  constructor(private firebaseAuth: AngularFireAuth,
+              private firebaseStorage: AngularFireStorage,
+              private toastr: ToastrService) {
     this.firebaseAuth.user.subscribe(
       user => this.user = user
     );
   }
 
   uploadFile(file: File) {
-    const storageRef = this.firebaseStorage.storage.ref();
-
-    storageRef.child(this.user.uid).child(Date.now() + '.' + file.name.split('.').pop()).put(file)
-      .then(snapshot => {
-        console.log('Uploaded a file!');
+    this.firebaseStorage.storage
+      .ref()
+      .child(this.user.uid)
+      .child(Date.now() + '.' + file.name.split('.').pop())
+      .put(file)
+      .then(() => {
```

Comment:
Would it not make more sense for the logic that happens once the file is uploaded to be in the class that called this function? (it can be moved by putting a 'return')

Or do you think it should be here?

Reply:
I think it should be in the storage service, but what you are saying makes sense as well. If I make use of the return where will I use it? Cause if I put it in the .then it might not execute when the previous logic fails. If I put the return outside the .then function it will execute without waiting for the .then to execute and finish.

Reply:
in front of `this.firebaseStorage.storage`, then you `.then()` in the calling class

Reply:
I also believe it can be in the service, cause this is logic that calls the toast api to render the toast view. This means that I am not responsible for rendering the view I only call their API to render it for me. Which for me means that it should lie in a service file.